src/Core/Checkout/Cart/Rule/CartVolumeRule.php

```diff
@@ -33,7 +35,11 @@ public function match(RuleScope $scope): bool
         }
 
         if ($this->volume === null) {
-            throw new UnsupportedValueException(\gettype($this->volume), self::class);
+            if (!Feature::isActive('v6.8.0.0')) {
+                // @phpstan-ignore-next-line
+                throw new UnsupportedValueException(\gettype($this->volume), self::class);
+            }
+            throw CartException::unsupportedValue(\gettype($this->volume), self::class);
```

Comment:
if I see it right, `CartException::unsupportedValue()` already returns the old Exception before 6.8. So we don't need the logic here. Won't comment every time, please check all.

Reply:
`CartException::unsupportedValue()` only returns self. Changing it is a BC.

```
    public static function unsupportedValue(string $type, string $class): self
    {
        return new self(
            Response::HTTP_BAD_REQUEST,
            self::VALUE_NOT_SUPPORTED,
            'Unsupported value of type {{ type }} in {{ class }}',
            ['type' => $type, 'class' => $class]
        );
    }
```

Reply:
whoops, confused it with the one in `LanguageException` 🙈 DomainExceptions ...